applications/nrf5340_audio/src/bluetooth/bt_stream/unicast/unicast_server.c

```diff
@@ -445,40 +434,66 @@ static int adv_buf_put(struct bt_data *adv_buf, uint8_t adv_buf_vacant, int *ind
 	return 0;
 }
 
-int unicast_server_config_get(uint32_t *bitrate, uint32_t *sampling_rate_hz,
-			      uint32_t *pres_delay_us)
+int unicast_server_config_get(struct bt_conn *conn, enum bt_audio_dir dir, uint32_t *bitrate,
+			      uint32_t *sampling_rate_hz, uint32_t *pres_delay_us)
 {
 	if (bitrate == NULL && sampling_rate_hz == NULL && pres_delay_us == NULL) {
 		LOG_ERR("No valid pointers received");
 		return -ENXIO;
 	}
 
-	if (audio_streams[0].codec_cfg == NULL) {
-		LOG_ERR("No codec found for the stream");
-		return -ENXIO;
-	}
+	if (dir == BT_AUDIO_DIR_SINK) {
+		if (audio_streams[0].codec_cfg == NULL) {
+			LOG_ERR("No codec found for the stream");
+			return -ENXIO;
+		}
 
-	if (sampling_rate_hz != NULL) {
-		*sampling_rate_hz = bt_audio_codec_cfg_get_freq(audio_streams[0].codec_cfg);
-	}
+		if (sampling_rate_hz != NULL) {
+			*sampling_rate_hz = bt_audio_codec_cfg_get_freq(audio_streams[0].codec_cfg);
+		}
 
-	if (bitrate != NULL) {
-		/* Get the configuration for the sink stream */
-		int frames_per_sec = 1000000 / bt_audio_codec_cfg_get_frame_duration_us(
-						       audio_streams[0].codec_cfg);
-		int bits_per_frame =
-			bt_audio_codec_cfg_get_octets_per_frame(audio_streams[0].codec_cfg) * 8;
+		if (bitrate != NULL) {
+			/* Get the configuration for the sink stream */
+			int frames_per_sec = 1000000 / bt_audio_codec_cfg_get_frame_duration_us(
+							       audio_streams[0].codec_cfg);
+			int bits_per_frame = bt_audio_codec_cfg_get_octets_per_frame(
+						     audio_streams[0].codec_cfg) *
+					     8;
 
-		*bitrate = frames_per_sec * bits_per_frame;
-	}
+			*bitrate = frames_per_sec * bits_per_frame;
+		}
+
+		if (pres_delay_us != NULL) {
+			if (audio_streams[0].qos == NULL) {
+				LOG_ERR("No QoS found for the stream");
+				return -ENXIO;
+			}
+
+			*pres_delay_us = audio_streams[0].qos->pd;
+		}
 
-	if (pres_delay_us != NULL) {
-		if (audio_streams[0].qos == NULL) {
-			LOG_ERR("No QoS found for the stream");
+	} else if (dir == BT_AUDIO_DIR_SOURCE) {
+#if CONFIG_BT_AUDIO_TX
```

Comment:
Does this have to be a precompiler check?

Reply:
I though it had to because of CONFIG_BT_ASCS_ASE_SRC_COUNT, but it seems it is used outside of precompiler check as well, so will try to remove as many as possible